### PR TITLE
checkout

### PR DESCRIPTION
**Pull request by dmtrung14**

### CODE REVIEW

File: controller/cartController.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
+import { BaseController } from "./baseController";
+const stripe = require('stripe')(process.env.STRIPE_SECRET_KEY);
```
Comment (Khiem17204) [inline controller/cartController.js:2]: import stripe from 'stripe';

cho consistent ES6

File: controller/cartController.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                currency: 'usd',
+                destination: 'KFC_STRIPE_ACCOUNT_ID',
+                description: 'Transfer for KFC order',
+              });
+
+            res.status(200).send('Payment successful');
+        
+        }
+        catch (error) {
+            res.status(500).send('Error processing payment');
```
Comment (Khiem17204) [inline controller/cartController.js:109]: 4xx, 500 la internal server error

File: controller/cartController.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+          .filter( key => predicate(obj[key]) )
+          .reduce( (res, key) => (res[key] = obj[key], res), {} );
+
+export class CartController extends BaseController {
+    constructor() {
+        this.bag = localStorage.getItem('bag') ? localStorage.getItem('bag') : {};
+        this.total = localStorage.getItem('price') ? localStorage.getItem('price') : 0;
+    }
+
+    updateLocalStorage() {
```
Comment (Khiem17204) [inline controller/cartController.js:15]: local storage tren backend thi la share giua cac thread day. ben js khong support parallel nen moi thg hit any endpoint la no tao 1 thread moi -> local storage share. Change to session if possible. khong thi ksao minh cung demo 1 lan thoi =))))

File: controller/cartController.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
+import { BaseController } from "./baseController";
+const stripe = require('stripe')(process.env.STRIPE_SECRET_KEY);
+
+Object.filter = (obj, predicate) =>
```
Comment (Khiem17204) [inline controller/cartController.js:4]: sao doi filter thanh cgi ki cuc v =)))) tao helper function di

File: controller/cartController.js [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+        this.bag = localStorage.getItem('bag') ? localStorage.getItem('bag') : {};
+        this.total = localStorage.getItem('price') ? localStorage.getItem('price') : 0;
+    }
+
+    updateLocalStorage() {
+        localStorage.setItem('price', Object.keys(this.bag).reduce((acc, key) => acc + this.bag[key].price * this.bag[key].amount, 0));
+        localStorage.setItem('bag', this.bag);
+    }
+     // Implement methods specific to the shopping bag controller here
+    update(req, res) {
```
Comment (Khiem17204) [inline controller/cartController.js:20]: so luong nua, nghi cach khac hoac implement 1 shopping cart class de handle cac case

**Review by Khiem17204 — COMMENTED** [review]
dep trai

File: controller/cartController.js (hunk 4 above)
Comment (dmtrung14) [reply controller/cartController.js:4]: này là helper function cho filter định nghĩa trên Object prototype mà bro. Bthg object (dict) thì đâu dùng filter được

File: controller/cartController.js (hunk 5 above)
Comment (dmtrung14) [reply controller/cartController.js:20]: so luong what?

File: controller/cartController.js (hunk 3 above)
Comment (dmtrung14) [reply controller/cartController.js:15]: the doi sang `sessionStorage` nhe?

File: controller/cartController.js (hunk 5 above)
Comment (Khiem17204) [reply controller/cartController.js:20]: so luong cua do chu sao, t map dit t an 2 cai burger

Comment (dmtrung14) [reply controller/cartController.js:20]: this.bag = {
item.id : {item.price, item.amount}
}

**Review by Khiem17204 — APPROVED** [review]
dep trai.

**Pull request merged**